### PR TITLE
Try to avoid duplicate profiling data

### PR DESCRIPTION
**Pull request by bantonsson**
The changes in #2587 shortened the diff between the `end` time in the `RecordingData` from the `snapshot` call and the next `start` time in the following `snapshot` call. Since the filtering of the `RecordingInputStream` that happens inside the JFR code will include chunks that started but didn't finish before `start`, and chunks that started but didn't finish before `end`, we could get some of the same chunks in consecutive snapshots that would lead to double counting.

This change tries to minimize the risk of getting the same chunks by using the actual time from the underlying `snapshot` recording as the `end` time of the `RecordingData` and add a small offset as the next `start` time.

Since the other change has not been released, there should not be a release note for this.

### CODE REVIEW

**Comment by jbachorik** [comment]
Nice catch with the recording stop time!
I think it would make sense to to fill the `end` timestamp of the `RecordingData` instance with the actual `end` timestamp retrieved from the JFR recording (eg. `Recording.getStopTime()`). This would propagate the 'true' end timestamp also to profiling backend which could use this information to eg. normalize reported values to the profile duration. 
What do you think?

**Comment by bantonsson** [comment]
Yes, that was my first thought as well, but then I hoped that this approach wouldn't break all the intrusive mock tests, but it did anyway. I'll change to that approach instead.

I still don't know what to do about the `OracleJdkRecordingData`, since the helper won't give the stop time.

File: dd-java-agent/agent-profiling/profiling-controller-oracle/src/main/java/com/datadog/profiling/controller/oracle/OracleJdkOngoingRecording.java [hunk 1]
```diff
@@ -56,7 +56,9 @@ public OracleJdkRecordingData snapshot(@Nonnull final Instant start, @Nonnull fi
       }
 
       targetName = helper.cloneRecording(targetName);
-      return new OracleJdkRecordingData(name, targetName, start, end, helper);
+      // Set the end time of the data to now, which is after the snapshot was made
+      // TODO get the recording end from the helper?
```
Comment (jbachorik) [inline dd-java-agent/agent-profiling/profiling-controller-oracle/src/main/java/com/datadog/profiling/controller/oracle/OracleJdkOngoingRecording.java:60]: I will take a look at what is possible to get from the Oracle JFR MX bean

**Review by richardstartin — APPROVED** [review]

File: dd-java-agent/agent-profiling/profiling-controller-oracle/src/main/java/com/datadog/profiling/controller/oracle/OracleJdkRecordingData.java [hunk 2]
```diff
@@ -23,9 +23,13 @@
 import java.util.Date;
 import javax.annotation.Nonnull;
 import javax.management.ObjectName;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /** Implementation for profiling recordings. */
 public class OracleJdkRecordingData extends RecordingData {
+  private static final Logger log = LoggerFactory.getLogger(OracleJdkRecordingData.class);
```
Comment (devinsba) [inline dd-java-agent/agent-profiling/profiling-controller-oracle/src/main/java/com/datadog/profiling/controller/oracle/OracleJdkRecordingData.java:31]: Looks like you don't need this anymore

**Review by tylerbenson — APPROVED** [review]

**Review by jbachorik — APPROVED** [review]

**Review by devinsba — APPROVED** [review]

**Pull request merged**